### PR TITLE
tests/ISSUE-103_accessibility_tests

### LINKED ISSUE #103: Accessibility tests
## ✏️ Description

Given the opportunity to write accessibility tests in Playwright, I propose to expand our tests to include accessibility tests of the GAD application.

## 🔧 Why We Need This

diversifying our portfolio
expanding our knowledge of accessibility testing

## 🚀 Proposed Solution

Accessibility tests for:
- welcome page
- main page
- Let's check button
- Comments button

## ✅ Issue reporting checklist

- [ ] I created a descriptive title for this issue
- [ ] I checked that it hasn't been reported yet
- [ ] I assigned the issue or added `help-wanted` label
- [ ] I added appropriate labels

### PR DESCRIPTION
**Pull request by Justyna-KO**
### Scope of changes:

Accessibility tests for:
-welcome page
-main page
-Let's check button
-Comments button

### How to use it:

  install @axe-core/playwright

## Any other comments?

Before submitting Pull Request, I've ensured that:

- [x] The PR is associated with the relevant Issue before its creation.
- [x] All new and existing tests passed
- [x] My commit messages follow the `kawqa-gad-playwright` project's [git commit message format](https://github.com/kat-kan/kawqa-gad-playwright/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Review by kat-kan — DISMISSED** [review]
I believe this is a good start for accessibility tests solution in our project.
IMO the target solution should be directed at using Page Object Pattern and be consistent with (not existing at the moment) UI tests solution.
Another thing that we could consider in the future is finding the alternative for skipping tests that are failing but we still want to keep them, Maybe exclude them by tag?  
But I believe these are not things that we can address right now so approve from my side :)

File: tests/accessibility/welcome-page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  expect(axeBuilder.violations).toEqual([]);
+});
+
+//For the purpose of the next test, I chose two standards that the page meets (wcag21a and wcag21aa).
+//A description of the standards can be found at: https://www.deque.com/axe/core-documentation/api-documentation/
+test('Check the whole site against selected standards', async ({ page }) => {
+  await page.goto(`/`);
+  await page.locator('#btnGui').waitFor();
+  const axeBuilder = await new AxeBuilder({ page })
+    .withTags('wcag21a', 'wcag21aa')
```
Comment (Slawomir-DKl) [inline tests/accessibility/welcome-page.spec.ts:18]: According to [AXE-Core Readme](https://github.com/dequelabs/axe-core-npm/blob/develop/packages/playwright/README.md), the option `withTags` requires string or array, ie. here it should be `.withTags(['wcag21a', 'wcag21aa'])` (I use "@axe-core/playwright": "^4.10.0")

File: tests/accessibility/main-page.spec.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+});
+
+//The test is skipped because the main page does not meet many accessibility standards and the test fails. You can run the test to generate a report with detailed information. At the end of the report (scroll down the error report) there is an additional attachment containing the results of the accessibility scan in json format.
+
+test.skip('Check the main page', async ({ page }, testInfo) => {
+  await page.goto(`/`);
+  await page.locator('#btnGui').waitFor();
+  await page.locator('#btnGui').click();
+  const axeBuilder = await new AxeBuilder({ page }).analyze();
+  await testInfo.attach('accesibility-scan-results', {
```
Comment (Slawomir-DKl) [inline tests/accessibility/main-page.spec.ts:25]: A typo - `accesibility` => `accessibility`

File: package.json [hunk 3]
```diff
@@ -27,5 +27,8 @@
     "style:check": "npx prettier . --check",
     "lint": "npx eslint . --ext .ts --max-warnings=0",
     "lint:fix": "npx eslint -- --fix"
+  },
+  "dependencies": {
+    "@axe-core/playwright": "^4.9.1"
```
Comment (Slawomir-DKl) [inline package.json:32]: The current version is 4.10.0 - maybe it would be good to have a current version as for the merge time

**Review by Slawomir-DKl — CHANGES_REQUESTED** [review]
Great experience - sth definitely worth learning. Thank you!

File: package.json (hunk 3 above)
Comment (kat-kan) [reply package.json:32]: We can update, but I'd like to note that it is not a general practice to update to latest version of library before the merge as changes were created and tested for the exact version. Here the scope is small so risk is low, but you can imagine what can be caused by rushing library update for bigger chunk of code :)

Comment (Justyna-KO) [reply package.json:32]: If this is not general practice, I would leave it as is.

File: tests/accessibility/main-page.spec.ts (hunk 2 above)
Comment (Justyna-KO) [reply tests/accessibility/main-page.spec.ts:25]: I changed it.

File: tests/accessibility/welcome-page.spec.ts (hunk 1 above)
Comment (Justyna-KO) [reply tests/accessibility/welcome-page.spec.ts:18]: I've corrected it.

File: package.json (hunk 3 above)
Comment (Slawomir-DKl) [reply package.json:32]: I haven't thought about the potential problems. Of course, you're right

**Review by Slawomir-DKl — DISMISSED** [review]
Thank you for the changes.

**Review by Slawomir-DKl — APPROVED** [review]

**Pull request merged**